examples/pbr/main.js

```diff
@@ -0,0 +1,1411 @@
+            target: [ 80.0, 0.0, 80.0 ],
+            eye: [ 80.0, -155.0, 120.0 ]
+        },
+        CameraMetal: {
+            target: [ 80.0, 0.0, 40.0 ],
+            eye: [ 80.0, -155.0, 80.0 ]
+        },
+        CameraCenter: {
+            target: [ 80.0, 0.0, 20.0 ],
+            eye: [ 80.0, -215.0, 20.0 ]
+        },
+        CameraPBR: {
+            target: [ 160.0, 0.0, 80.0 ],
+            eye: [ 160.0, -100.0, 80.0 ]
+        },
+        CameraSamples: {
+            target: [ 46.0, 20.0, 80.0 ],
+            eye: [ 46.0, -62.5, 80.0 ]
+        }
+
+
+    };
+
+
+    var isMobileDevice = function () {
+
+        if ( navigator.userAgent.match( /Mobile/i ) )
+            return true;
+        if ( navigator.userAgent.match( /Android/i ) )
+            return true;
+        if ( navigator.userAgent.match( /iPhone/i ) )
+            return true;
+        if ( navigator.userAgent.match( /iPad/i ) )
+            return true;
+        if ( navigator.userAgent.match( /iPod/i ) )
+            return true;
+        if ( navigator.userAgent.match( /BlackBerry/i ) )
+            return true;
+        if ( navigator.userAgent.match( /Windows Phone/i ) )
+            return true;
+
+        return false;
+
+    };
+
+
+    var optionsURL = {};
+    ( function ( options ) {
+        var vars = [],
+            hash;
+        var indexOptions = window.location.href.indexOf( '?' );
+        if ( indexOptions < 0 ) return;
+
+        var hashes = window.location.href.slice( indexOptions + 1 ).split( '&' );
+        for ( var i = 0; i < hashes.length; i++ ) {
+            hash = hashes[ i ].split( '=' );
+            var element = hash[ 0 ];
+            vars.push( element );
+            var result = hash[ 1 ];
+            if ( result === undefined ) {
+                result = '1';
+            }
+            options[ element ] = result;
+        }
+    } )( optionsURL );
+
+
+    var linear2Srgb = function ( value, gammaIn ) {
+        var gamma = gammaIn;
+        if ( !gamma ) gamma = 2.2;
+        var result = 0.0;
+        if ( value < 0.0031308 ) {
+            if ( value > 0.0 )
+                result = value * 12.92;
+        } else {
+            result = 1.055 * Math.pow( value, 1.0 / gamma ) - 0.055;
+        }
+        return result;
+    };
+
+    var shaderProcessor = new osgShader.ShaderProcessor();
+
+    window.ALBEDO_TEXTURE_UNIT = 2;
+    window.DIFFUSE_TEXTURE_UNIT = 2;
+    window.ROUGHNESS_TEXTURE_UNIT = 3;
+    window.METALNESS_TEXTURE_UNIT = 4;
+    window.NORMAL_TEXTURE_UNIT = 5;
+    window.SPECULAR_TEXTURE_UNIT = 4;
+
+
+    window.formatList = [ 'FLOAT', 'RGBE', 'RGBM', 'LUV' ];
+
+    var modelsPBR = [ 'cerberus', 'c3po', 'devastator' ];
+
+    var modelList = [ 'sphere', 'model' ];
+    if ( window.useExternalModels ) {
+        modelList = modelList.concat( modelsPBR );
+    }
+
+
+    var Example = function () {
+
+        this._shaderPath = 'shaders/';
+
+        if ( optionsURL.colorEncoding )
```

Comment:
Could derive from Example, if only to get OptionsURL for free as it's now part of Example, and the parameters ends in Example::_config.

(but surely can get other things like shaderprocessor, shader loads, etc)